nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncodeContent.java

```diff
@@ -55,27 +59,105 @@
 @CapabilityDescription("Encode or decode the contents of a FlowFile using Base64, Base32, or hex encoding schemes")
 public class EncodeContent extends AbstractProcessor {
 
-    public static final String ENCODE_MODE = "Encode";
-    public static final String DECODE_MODE = "Decode";
-
-    public static final String BASE64_ENCODING = "base64";
-    public static final String BASE32_ENCODING = "base32";
-    public static final String HEX_ENCODING = "hex";
+    public static final AllowableValue ENCODE_MODE = new AllowableValue("encode", "Encode", "Sets the operation mode to 'encode'.");
+    public static final AllowableValue DECODE_MODE = new AllowableValue("decode", "Decode", "Sets the operation mode to 'decoding'.");
+
+    public static final AllowableValue BASE64_ENCODING = new AllowableValue("base64", "Base64", "Sets the encoding type to 'Base64'.");
+    public static final AllowableValue BASE32_ENCODING = new AllowableValue("base32", "Base32", "Sets the encoding type to 'Base32'.");
+    public static final AllowableValue HEX_ENCODING = new AllowableValue("hex", "Hexadecimal", "Sets the encoding type to 'Hexadecimal'.");
```

Comment:
Instead of defining AllowableValue instances, it would be helpful to define an enum that implements `DescribedValue`, which makes it easier to use the property value during processing.

Reply:
Done.  added

1. nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/encoding/EncodingMode.java
2. nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/encoding/EncodingTypejava
3. nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/encoding/LineOutputMode.java